### PR TITLE
Adding Docs + Publishing NPM Script + README Tweaks

### PR DESCRIPTION
**Pull request by gauntface**
This adds a publishing step and adds doc generation to the project.

This was largely kicked off by: https://github.com/GoogleChrome/push-encryption-node/issues/22

@addyosmani @jeffposnick @surma would one of you kind souls give this PR a review?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,15 +11,19 @@
...
-    "sinon": "^1.17.3"
+    "require-dir": "^0.3.0",
+    "sinon": "^1.17.3",
+    "sw-testing-helpers": "0.0.12"
   },
   "scripts": {
     "test": "gulp lint test:manual",
-    "build-docs": "./node_modules/jsdoc/jsdoc.js -c jsdoc.json"
+    "build-docs": "jsdoc -c jsdoc.json",
+    "bundle": "./project/copy-bundle-files.sh",
```
Comment (jeffposnick) [inline package.json:24]: Shouldn't this pass in a directory as a command line parameter to the shell script?

File: package.json [hunk 2]
```diff
@@ -11,15 +11,19 @@
...
+    "publish": "./node_modules/sw-testing-helpers/project/publish-release.sh",
+    "build": "echo \"Skipping build step\""
```
Comment (jeffposnick) [inline package.json:26]: What's the point of this?

File: package.json (hunk 1 above)
Comment (gauntface) [inline package.json:24]: It does, but that argument is passed to the npm run script that is then passed to the shell script.

If anything goes wrong, the shell script will catch it and exit.

File: package.json (hunk 2 above)
Comment (gauntface) [inline package.json:26]: The re-usable shell script expects this to exist at the moment I'm intending on changing this as soon as possible but for now this is the easiest solution.

See issue: https://github.com/GoogleChrome/sw-testing-helpers/issues/21

**Comment by jeffposnick** [comment]
:+1:

**Pull request merged**